pkg/workload/schemachange/generate_test.go

```diff
@@ -0,0 +1,165 @@
+	r := rand.New(rand.NewSource(0))
+	funcs := map[string]any{
+		"Database": func() *tree.Name {
+			db := tree.Name("MyDatabase")
+			return &db
+		},
+		"ReturnErrCaseNotPossible": func() (tree.Name, error) {
+			return "", ErrCaseNotPossible
+		},
+	}
+
+	t.Run("Successful", func(t *testing.T) {
+		// Successful cases will be prioritized when they are possible and
+		// produceError is false.
+		stmt, code, err := Generate[*tree.RenameDatabase](r, false, []GenerationCase{
+			{pgcode.SuccessfulCompletion, `{ReturnErrCaseNotPossible}`},
+			{pgcode.SuccessfulCompletion, `ALTER DATABASE {Database} RENAME TO NewName`},
+		}, funcs)
+		require.NoError(t, err)
+		require.Equal(t, pgcode.SuccessfulCompletion, code)
+		require.Equal(t, &tree.RenameDatabase{
+			Name:    "MyDatabase",
+			NewName: "newname",
+		}, stmt)
+	})
+
+	t.Run("SyntaxError", func(t *testing.T) {
+		_, _, err := Generate[*tree.RenameDatabase](r, false, []GenerationCase{
+			{pgcode.SuccessfulCompletion, `ALTER DATABASE {Database}`},
+		}, funcs)
+		require.EqualError(t, err, `syntax error; could not parse *tree.RenameDatabase: "ALTER DATABASE \"MyDatabase\""`)
+	})
+
+	t.Run("WrongType", func(t *testing.T) {
+		_, _, err := Generate[*tree.RenameDatabase](r, false, []GenerationCase{
+			{pgcode.SuccessfulCompletion, `ALTER TABLE foo RENAME TO bar`},
+		}, funcs)
+		require.EqualError(t, err, `expected to parse *tree.RenameDatabase; got *tree.RenameTable: "ALTER TABLE foo RENAME TO bar"`)
+	})
+
+	t.Run("NoCasesPossible", func(t *testing.T) {
+		_, _, err := Generate[*tree.RenameDatabase](r, false, nil, funcs)
+		require.ErrorIs(t, err, ErrCaseNotPossible)
+
+		_, _, err = Generate[*tree.RenameDatabase](r, false, []GenerationCase{
+			{pgcode.SuccessfulCompletion, `{ReturnErrCaseNotPossible}`},
+		}, funcs)
+		require.ErrorIs(t, err, ErrCaseNotPossible)
+
+		// Cases will be tried until we either exhaust all cases or we find a valid
+		// case.
+		_, _, err = Generate[*tree.RenameDatabase](r, false, []GenerationCase{
+			{pgcode.SuccessfulCompletion, `{ReturnErrCaseNotPossible}`},
+			{pgcode.SuccessfulCompletion, `{ReturnErrCaseNotPossible}`},
+			{pgcode.SuccessfulCompletion, `{ReturnErrCaseNotPossible}`},
+			{pgcode.SuccessfulCompletion, `ALTER DATABASE foo RENAME TO bar`},
+		}, funcs)
+		require.NoError(t, err)
+	})
+
+	t.Run("ProduceError", func(t *testing.T) {
+		// If the only cases provided are successes and produceError is true, we'll
+		// error out with case not possible.
+		_, _, err := Generate[*tree.RenameDatabase](r, true, []GenerationCase{
+			{pgcode.SuccessfulCompletion, `ALTER DATABASE {Database} RENAME TO NewName`},
+			{pgcode.SuccessfulCompletion, `ALTER DATABASE {Database} RENAME TO NewName`},
+			{pgcode.SuccessfulCompletion, `ALTER DATABASE {Database} RENAME TO NewName`},
+		}, funcs)
+		require.ErrorIs(t, err, ErrCaseNotPossible)
+
+		// Assert that only error cases will be selected if produceError is true.
+		for i := 0; i < 100; i++ {
```

Comment:
nit: what's the point of iterating 100 times?

Reply:
Because these tests utilize rng, I wanted to get more reps in to indicate that the test had value. I've replaced them with `quick.Check` calls instead.